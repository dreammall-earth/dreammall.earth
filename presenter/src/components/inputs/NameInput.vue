<template>
  <v-text-field
    ref="inputField"
    :model-value="props.modelValue"
    :name="props.name"
    :class="props.class"
    :label="props.label"
    :variant="props.variant"
    :color="props.color"
    :bg-color="props.bgColor"
    :hide-details="props.hideDetails"
    :flat="props.flat"
    :rounded="props.rounded"
    :type="props.type"
    :required="props.required"
    :rules="nameRules"
    @update:model-value="emit('update:modelValue', $event)"
  ></v-text-field>
</template>

<script lang="ts" setup>
import { ref } from 'vue'

import { nameRules } from '#src/validation/validation'

const props = withDefaults(
  defineProps<{
    modelValue: string | null
    name?: string
    class?: string
    label?: string
    variant?:
      | 'filled'
      | 'outlined'
      | 'plain'
      | 'underlined'
      | 'solo'
      | 'solo-inverted'
      | 'solo-filled'
      | undefined
    color?: string
    bgColor?: string
    hideDetails?: boolean | 'auto' | undefined
    flat?: boolean
    rounded?: string | boolean
    type?: string
    required?: boolean
  }>(),
  {
    name: undefined,
    class: undefined,
    label: undefined,
    variant: 'solo',
    color: '#3D4753',
    bgColor: 'rgba(174, 179, 189, 0.50)',
    hideDetails: 'auto',
    flat: true,
    rounded: 'xl',
    type: 'text',
    required: false,
  },
)

const emit = defineEmits<{
  (e: 'update:modelValue', event: string): void
}>()

const inputField = ref()
const focusInput = () => {
  if (inputField.value) {
    inputField.value.focus()
  }
}
defineExpose({
  inputField,
  focusInput,
})
</script>
