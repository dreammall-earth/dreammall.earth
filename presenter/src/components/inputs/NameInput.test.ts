import { mount } from '@vue/test-utils'
import { describe, it, expect, beforeEach } from 'vitest'

import NameInput from './NameInput.vue'

describe('MainButton', () => {
  const Wrapper = () => {
    return mount(NameInput, { props: { modelValue: '' } })
  }
  let wrapper: ReturnType<typeof Wrapper>

  beforeEach(() => {
    wrapper = Wrapper()
  })

  it('renders', () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  it('emits the update:modelValue event on input', async () => {
    await wrapper.find('input').setValue('New Name')
    expect(wrapper.emitted('update:modelValue')).toEqual([['New Name']])
  })
})
