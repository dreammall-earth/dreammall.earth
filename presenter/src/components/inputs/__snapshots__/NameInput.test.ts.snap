// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MainButton > renders 1`] = `
<div
  class="v-input v-input--horizontal v-input--center-affix v-input--density-default v-locale--is-ltr v-text-field"
>
  <!---->
  <div
    class="v-input__control"
  >
    
    <div
      class="v-field v-field--center-affix v-field--flat v-field--has-background v-field--no-label v-field--variant-solo v-theme--light rounded-xl v-locale--is-ltr"
      style="background-color: rgba(174, 179, 189, 0.50);"
    >
      <div
        class="v-field__overlay"
      />
      <div
        class="v-field__loader"
      >
        <div
          aria-hidden="true"
          aria-valuemax="100"
          aria-valuemin="0"
          class="v-progress-linear v-theme--light v-locale--is-ltr"
          role="progressbar"
          style="top: 0px; height: 0px; --v-progress-linear-height: 2px; left: 50%; transform: translateX(-50%);"
        >
          <!---->
          <div
            class="v-progress-linear__background"
            style="background-color: #3D4753; color: #fff; caret-color: #fff; width: 100%;"
          />
          <transition-stub
            appear="false"
            css="true"
            name="fade-transition"
            persisted="false"
          >
            <div
              class="v-progress-linear__indeterminate"
            >
              
              <div
                class="v-progress-linear__indeterminate long"
                style="background-color: #3D4753; color: #fff; caret-color: #fff;"
              />
              <div
                class="v-progress-linear__indeterminate short"
                style="background-color: #3D4753; color: #fff; caret-color: #fff;"
              />
              
            </div>
          </transition-stub>
          <!---->
        </div>
      </div>
      <!---->
      <div
        class="v-field__field"
        data-no-activator=""
      >
        <!---->
        <label
          class="v-label v-field-label"
          for="input-0"
        >
          <!---->
          
          <!---->
          
        </label>
        
        
        <!---->
        <input
          aria-describedby="input-0-messages"
          class="v-field__input"
          id="input-0"
          size="1"
          type="text"
        />
        <!---->
        
        
      </div>
      <!---->
      <!---->
      <div
        class="v-field__outline"
      >
        <!---->
        <!---->
      </div>
    </div>
    
  </div>
  <!---->
  <!---->
</div>
`;
